src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -346,7 +346,7 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                         // All primary constructor parameters are definitely assigned outside of the primary constructor
                         foreach (var parameter in primaryConstructor.Parameters)
                         {
-                            NoteWrite(parameter, value: null, read: true);
+                            NoteWrite(parameter, value: null, read: true, isRef: false);
```

Comment:
Does this behave correctly when a primary constructor parameter is by-`ref`? For example, do ref parameters in primary constructors behave consistently with by-value parameters in the dataflow APIs?

Reply:
I don't think this should affect dataflow APIs in any way, since the `isRef` is only passed down to `_sourceAssembly.NoteFieldAccess` which isn't part of dataflow but is only used to report the "unused field" diagnostics IIUC. Moreover, it's only used if the symbol is a field but here it's a parameter, so that branch won't be reached.

That being said, in case we used the parameter elsewhere in the future, we could probably pass `isRef: parameter.RefKind != RefKind.None` here.